instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -615,6 +620,18 @@ def _get_otel_send(
     ):
         @wraps(send)
         async def otel_send(message):
+            if message.get("headers"):
```

Comment:
Can you move this logic to a new function?
This logic is not really part of otel_send logic, so I think it will be better to keep the function simple

Reply:
I refactored the implementation, and now the main logic belongs to the correct place. Thanks for your comment.